service/worker/migration/activities.go

```diff
@@ -474,9 +490,38 @@ func (a *activities) ListWorkflows(ctx context.Context, request *workflowservice
 	}
 	var lastCloseTime, lastStartTime time.Time
 
-	executions := make([]*commonpb.WorkflowExecution, len(resp.Executions))
-	for i, e := range resp.Executions {
-		executions[i] = e.Execution
+	executions := make([]*replicationspb.MigrationExecutionInfo, 0, len(resp.Executions))
+	for _, e := range resp.Executions {
+		executionInfo := &replicationspb.MigrationExecutionInfo{
+			WorkflowId: e.Execution.GetWorkflowId(),
+			RunId:      e.Execution.GetRunId(),
+			// Ideally we should use chasm.WorkflowArchetypeID by default,
+			// but for backward compatibility reason we need this field to be 0
+			// to avoid unmarshaling errors since the previous type returned in
+			// ListWorkflowExecutions.Executions is commonpb.WorkflowExecution
+			// which does not have ArchetypeId field.
+			//
+			// TODO: switch to chasm.WorkflowArchetypeID in release 1.31.0
+			ArchetypeId: chasm.UnspecifiedArchetypeID,
+		}
+
+		nsDivisionPayload, ok := e.SearchAttributes.IndexedFields[sadefs.TemporalNamespaceDivision]
```

Comment:
Is this ensured to not be nil?

Reply:
Updated to use GetIndexedFields() and performs nil check on the map.